spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j/AmqpAppender.java

```diff
@@ -409,6 +409,8 @@ public void setCharset(String charset) {
 	 */
 	protected void startSenders() {
 		senderPool = Executors.newCachedThreadPool();
+		synchronized(this) {
+		} // (logically) flush all variables to main memory
```

Comment:
H-m-m... Thanks for the explanation.
Don't we have anything similar with our `EnableIntegrationTests` and `InterceptedSharedConnectionTests`?